vsphere/resource_vsphere_compute_cluster.go

```diff
@@ -1478,15 +1507,8 @@ func resourceVSphereComputeClusterApplyVsanConfig(d *schema.ResourceData, meta i
 		},
 	}
 
-	dedupEnabled := d.Get("vsan_dedup_enabled").(bool)
-	compressionEnabled := d.Get("vsan_compression_enabled").(bool)
-	if dedupEnabled && !compressionEnabled {
-		return fmt.Errorf("vsan compression must be enabled if vsan dedup is enabled")
-	}
-
-	conf.DataEfficiencyConfig = &vsantypes.VsanDataEfficiencyConfig{
-		DedupEnabled:       dedupEnabled,
-		CompressionEnabled: &compressionEnabled,
+	if version.AtLeast(viapi.VSphereVersion{Product: version.Product, Major: 8, Minor: 0}) {
+		conf.VsanClusterConfig.(*vsantypes.VsanClusterConfigInfo).VsanEsaEnabled = structure.GetBool(d, "vsan_esa_enabled")
```

Comment:
If we opt for the non-nil implementation I've described you wouldn't need to use `GetBool` you would simply pass a pointer with an explicitly valued Boolean (see `compressionEnabled`)